analyses/chromosomal-instability/chromosomal-instability.Rmd

```diff
@@ -0,0 +1,387 @@
+
+  # Make the combo plot
+  multipanel_break_plot(
+    granges_list = granges_list,
+    plot_name = sample_name,
+    y_val = "total_counts",
+    y_lab = "Breaks per Mb",
+    plot_dir = sample_plots_dir
+  )
+})
+```
+
+### Co-localization of breakpoints for tumor-type groups
+
+Same as was done for each sample, now we will calculate densities for 
+each tumor-type group. 
+
+```{r, results='hide'}
+# Change bin_size here and it will change the rest
+bin_size <- 1e6
+
+# Get a list of the tumor_types for which we have DNA-seq data
+tumor_types <- metadata %>% 
+  dplyr::filter(!is.na(short_histology), experimental_strategy != "RNA-Seq") %>% 
+  dplyr::distinct(short_histology) %>%
+  dplyr::pull(short_histology)
+
+tumor_types[12]
+```
+
+Run the density calculations for the groups. 
+
+```{r}
+# Get a big list of break densities for each sample.
+group_densities <- lapply(tumor_types, function(tumor_type) {
+
+  # Obtain a list of sample_ids to subset by
+  sample_ids <- metadata %>%
+    dplyr::filter(metadata$short_histology == tumor_type) %>%
+    dplyr::pull(Kids_First_Biospecimen_ID)
+  
+  # Double check these samples are in the list
+  check_samples <- (sample_ids %in% common_samples)
+  
+  # If no samples, go to next
+  if (sum(check_samples) > 1) {
+  
+  message(paste("Calculating breakpoint density for", tumor_type, "samples"))
+
+    # Calculate break densities with both datasets combined
+  common_density <- break_density(
+    sv_breaks,
+    cnv_breaks,
+    sample_id = sample_ids,
+    window_size = bin_size,
+    chr_sizes_list = chr_sizes_list
+  )
+
+  # Calculate break densities for CNV only
+  cnv_density <- break_density(
+    cnv_breaks,
+    sample_id = sample_ids,
+    window_size = bin_size,
+    chr_sizes_list = chr_sizes_list
+  )
+
+  # Calculate break densities for SV only
+  sv_density <- break_density(
+    sv_breaks,
+    sample_id = sample_ids,
+    window_size = bin_size,
+    chr_sizes_list = chr_sizes_list
+  )
+
+  # Return this mega GRanges list
+  return(list(
+    common_density = common_density,
+    cnv_density = cnv_density,
+    sv_density = sv_density
+  ))
+  }
+})
+
+# Bring along the tumor-type labels
+names(group_densities) <- tumor_types
+
+# Remove tumor_types data that didn't have at least two samples
+group_densities <- group_densities[!sapply(group_densities, is.null)]
+```
+
+### Plot the breakpoints for each tumor-type
+
+Here we will plot median number of break points for the tumor-type group per 
+each bin.
+
+```{r}
+lapply(group_densities, function(granges_list) {
+  # Get the tumor type label
+  group_name <- tumor_types[parent.frame()$i[]]
+
+  # Make the combo plot
+  multipanel_break_plot(
+    granges_list = granges_list,
+    plot_name = group_name,
+    y_val = "median_counts",
+    y_lab = "Median Breaks per Mb",
+    plot_dir = hist_plots_dir
+  )
+})
+```
+
+Zip up the PNG files into one file. 
+
+```{r}
+zip(paste0(hist_plots_dir, ".zip"), 
```

Comment:
somehow this ended up with a folder within a folder... My guess is this happens if the zip file already exists from a previous run.

Reply:
This should be fixed with the most recent commit I did.